File: google-maps/android/src/main/java/com/capacitorjs/plugins/googlemaps/GoogleMapPadding.kt
package com.capacitorjs.plugins.googlemaps

import org.json.JSONObject

class GoogleMapPadding(fromJSONObject: JSONObject) {
    var top: Int = 0
    var bottom: Int = 0
    var left: Int = 0
    var right: Int = 0

    init {
        if(fromJSONObject.has("top")) {
            top = fromJSONObject.getInt("top")
        }

        if(fromJSONObject.has("bottom")) {
            bottom = fromJSONObject.getInt("bottom")
        }

        if(fromJSONObject.has("left")) {
            left = fromJSONObject.getInt("left")
        }

        if(fromJSONObject.has("right")) {
            right = fromJSONObject.getInt("right")
        }
    }
}

File: google-maps/ios/Sources/CapacitorGoogleMapsPlugin/GoogleMapPadding.swift
import Foundation
import Capacitor

public struct GoogleMapPadding {
    let top: Float
    let bottom: Float
    let left: Float
    let right: Float

    init(fromJSObject: JSObject) throws {
        top = fromJSObject["top"] as? Float ?? 0
        bottom = fromJSObject["bottom"] as? Float ?? 0
        left = fromJSObject["left"] as? Float ?? 0
        right = fromJSObject["right"] as? Float ?? 0
    }
}
